felix/dataplane/linux/bpf_ep_mgr.go

```diff
@@ -1117,11 +1118,18 @@ func (m *bpfEndpointManager) onInterfaceUpdate(update *ifaceStateUpdate) {
 					delete(m.initAttaches, update.Name)
 				}
 			}
+			if update.Name == dataplanedefs.BPFInDev {
+				m.ifacesLock.Lock()
+				if err := m.reconcileBPFDevices(dataplanedefs.BPFInDev); err != nil {
+					log.WithError(err).Fatal("Failed to configure BPF devices")
+				}
+				defer m.ifacesLock.Unlock()
```

Comment:
either use defer right after Lock or just Unlock here. The fatal above won't return, it will kill the process. Why don't you handle it below where you handle  Out?